sktime/transformations/base.py

```diff
@@ -269,6 +269,49 @@ def __radd__(self, other):
         else:
             return NotImplemented
 
+    def __getitem__(self, key):
+        """Magic [...] method, return column subsetted transformer.
+
+        First index does output subsetting, second index does input subsetting.
```

Comment:
hm, my thinking was that output subsetting is much more common, and in `pandas.DataFrame`, the syntax `t["a"]` would subset columns. Switching it around would make that inconsistent.

Reply:
Question - couldn't we just have a case where if there is only one argument passed it is assumed to be for output subsetting, but if there are 2 then the order is input, output?  Is that just more confusing/terrible design because the optional argument is pre-pended??  Again, not a blocker, so feel free not to follow up on this, just a thought

Reply:
we could alway try to hack the python language itself to allow for sth like `["input"]t["output"]`

Either way, this might be a nice design item to bring up in the Fr meeting.

Reply:
as discussed today, this is now using syntax `t["input", "output]` if two arguments are provided, and `t["output"]` if just one argument is provided.

Reply:
I'm changing this to `t["input", "output]` if two arguments are provided, and `t["input"]` if just one argument is provided, based on recent discussion at the dev days. Is that ok, @miraep8, @benheid, @smeisen, @kalebphipps?

We probably won't use this in graph pipeline construction, and that way it is easier to use with `FeatureUnion`.